pkg/cmd/eventlistener/portforward.go

```diff
@@ -0,0 +1,114 @@
+package eventlistener
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"strconv"
+
+	"github.com/spf13/cobra"
+	"github.com/tektoncd/cli/pkg/cli"
+	"github.com/tektoncd/cli/pkg/eventlistener"
+	"github.com/tektoncd/cli/pkg/formatted"
+	"github.com/tektoncd/cli/pkg/options"
+	"github.com/tektoncd/triggers/pkg/reconciler/eventlistener/resources"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+)
+
+var defaultPort = strconv.Itoa(resources.DefaultPort)
+
+func portforwardCommand(p cli.Params) *cobra.Command {
+	opts := &options.PortForwardOptions{}
+
+	eg := `Forward port '9001' from EventListener 'foo' in namespace 'bar' to localhost port '8001':
+
+	tkn eventlistener port-forward foo -n bar -p 8001:9001
+
+or
+
+	tkn el pf foo -n bar -p 8001:9001
+`
+
+	c := &cobra.Command{
+		Use:          "port-forward",
+		Aliases:      []string{"pf"},
+		Short:        "Port forwards an EventListener in a namespace",
+		Example:      eg,
+		SilenceUsage: true,
+		Annotations: map[string]string{
+			"commandType": "main",
+		},
+		Args:              cobra.ExactValidArgs(1),
+		ValidArgsFunction: formatted.ParentCompletion,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cs, err := p.Clients()
+			if err != nil {
+				return fmt.Errorf("failed to create tekton client")
+			}
+
+			pods, err := eventlistener.Pods(cs, args[0], metav1.GetOptions{}, p.Namespace())
```

Comment:
that will crash i think if there is no args on the command line,

Reply:
cobra will error out instead because it requires exactly 1 arg https://github.com/tektoncd/cli/blob/e4693b0dd32e00f309afdd141d05b4d9de67f6a6/pkg/cmd/eventlistener/portforward.go#L45